share/agrammon.openapi

```diff
@@ -369,6 +427,150 @@ components:
+                    type: string
+        Input:
+            type: object
+            additionalProperties: true
+            properties:
+                branch:
+                    type: string
+                    enum:
+                        - 'true'
+                        - 'false'
+                defaults:
+                    type: object
+                    properties:
+                        calc:
+                            nullable: true
+                            anyOf:
+                                - type: string
+                                - type: number
+                        gui:
+                            nullable: true
+                            anyOf:
+                                - type: string
+                                - type: number
+                enum:
+                    description: XXX
+                    type: object
+                    additionalProperties: true
+                gui:
+                    description: GUI navigation label
+                        schema:
+                            $ref: '#/components/schemas/LanguageLabels'
+                help:
+                    description: Help texts
+                        schema:
+                            $ref: '#/components/schemas/LanguageLabels'
+                labels:
+                    description: Input labels
+                        schema:
+                            $ref: '#/components/schemas/LanguageLabels'
+                models:
+                    description: Model variants this input is used for
+                    type: array
+                    items:
+                        type: string
+                options:
+                    description: XXX
+                    type: array
+                    items:
+                        type: array
+                        items:
+                            type: string
+                optionsLang:
+                    description: XXX
+                    type: array
+                    items:
+                        schema:
+                            $ref: '#/components/schemas/LanguageLabels'
+#                        type: object
+                order:
+                    description: Number used for sorting
+                    type: integer
+                type:
+                    description: Data type
+                    type: string
+                    enum:
+                        - 'float'
+                        - 'integer'
+                        - 'percent'
+                        - 'text'
+                        - 'enum'
+                        - 'enum{ignore}'
+                units:
+                    description: Input labels
+                        schema:
+                            $ref: '#/components/schemas/LanguageLabels'
+                validator:
+                    description: Input validator
+                    type: object
+# TODO: can be either empty hash or must have name and args
```

Comment:
@jnthn Do you know how this can be done in OpenAPI?

Reply:
I think you're looking for [`oneOf` or `anyOf`](https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/).

Reply:
The problem was, that I didn't know how to specify an empty hash/object, now I do:
```
EmptyHash:
    type: object
    properties: {}
```
Fixed.